### PR TITLE
[mqtt.generic] Support transformationPattern for thing's availabilityTopic

### LINKED ISSUE #10226: [mqtt] support availability transformation in generic mqtt thing
i have a few devices that send availability messages to some topic `/status` that are json and include a little bit of extra metadata:

ONLINE:

{"status":"ONLINE", "timestamp": .... }

OFFLINE:
{"status":"OFFLINE", ...}

oh3 doesn't let me extract the "status" field from the json so the thing always shows up as offline

### LINKED ISSUE #10632: [MQTT] enable transformationPattern for device's LWT / availabilityTopic (Mainly JSON)
Some off the shelf MQTT devices (that are not DIY), advertises their LW message in a non standard format. Need to allow transformationPattern for availabilityTopic in  MQTT's thing topic.

Add-on: MQTT

Example: an Advantech MQTT industrial IO advertises LWT in a JSON format:

availabilityTopic:
'Advantech/<MAC Address>/Device_Status'

LW (offline)
{"status":"disconnect","name":"ADAM6024-D","macid":"<MAC Address>","ipaddr":"<IP Address>"}
LW (online)
{"status":"onnect","name":"ADAM6024-D","macid":"<MAC Address>","ipaddr":"<IP Address>"}

Running Openhab 3.0.2:

MQTT thing:

```
Bridge mqtt:broker:Advantech "Advantech" [ host="127.0.0.1", port=1883, secure=false, clientID="Advantech" ]
{
    Thing topic ADAM6024D_<MAC Address>  "ADAM6024D_<MAC Address>" [availabilityTopic="Advantech/<MAC Address>/Device_Status", `transformationPattern="JSONPATH:$.status"`, payloadAvailable="connect", payloadNotAvailable="disconnect"
    {
        Channels:
        Type number: ai1        "AI1"           [stateTopic="Advantech/<MAC Address>/data", transformationPattern="JSONPATH:$.ai1"]

    }
}
```
the `transformationPattern="JSONPATH:$.status"` in the Thing Topic seems to be ignored, it does not extract the status, the log shows the that the JSON received is not a valid command.

The current workaround is to specify the payloadAvailable/NotAvailable as a whole string (including the uniqe MAC Address and IP Address) which is not efficient.

### PR DESCRIPTION
**Pull request by jimtng**
Resolve #10632, Resolve #10226 by adding transformationPattern option for Thing's availabilityTopic.
 
De-duplicated the transformation string parsing routine, so that all transformationPattern[Out] get parsed by the same code.

### CODE REVIEW

File: bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/ChannelState.java [hunk 1]
```diff
@@ -20,7 +20,9 @@
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
 
+import org.apache.commons.lang3.StringUtils;
```
Comment (lolodomo) [inline bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/ChannelState.java:25]: See #7722 
Can StringUtils.isNotBlank be replaced by  !isBlank?

**Review by lolodomo — CHANGES_REQUESTED** [review]
I am not involved at all in the MQTT code but can"t the OH transformation stuff be rather used?

File: bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/ChannelState.java (hunk 1 above)
Comment (jimtng) [reply bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/ChannelState.java]: Great idea! done.

Comment (jimtng) [reply bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/ChannelState.java]: > I am not involved at all in the MQTT code but can"t the usual OH transformation stuff be rather used?

Neither am I. I'm just happen to need some features from mqtt. From what I can see, it is using OH transformation under the hood:

https://github.com/openhab/openhab-addons/blob/882e52647da3dfddbf35e896368c29cadc8c8a92/bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/ChannelStateTransformation.java#L20-L33

Comment (jlaur) [reply bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/ChannelState.java]: @lolodomo, @jimtng - can this comment be considered resolved?

Comment (lolodomo) [reply bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/ChannelState.java]: Not yet @jlaur

Comment (lolodomo) [reply bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/ChannelState.java]: My question is in fact: is all the transformation addons provided by OH that can be applied by the user on any item state not sufficient?

Comment (lolodomo) [reply bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/ChannelState.java]: If you are sure this is the good approach, let say it is ok for me.

Comment (jimtng) [reply bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/ChannelState.java]: Do you have something different in mind?

**Comment by lolodomo** [comment]
There is not yet any default translation properties file.

**Review by lolodomo — APPROVED** [review]
LGTM

**Pull request merged**